src/libraries/System.Linq/src/System/Linq/Range.SpeedOpt.cs

```diff
@@ -2,31 +2,37 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace System.Linq
 {
     public static partial class Enumerable
     {
         private sealed partial class RangeIterator : IPartition<int>
         {
+            // Up to 16 elements which corresponds to AVX512
+            private static readonly Vector<int> InitMask = new(
```

Comment:
Just a note: it's usually better to inline Vector.Create at uses

Reply:
Apparently inline version as-is regresses the codegen by a few extra `mov`s and a `cmp`.
```cs
Vector<int> initMask = new(
    (ReadOnlySpan<int>)(ReadOnlySpan<int>)new int[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 })
```
but if initialized with the below the regression goes away and becomes the same as the original code
```cs
Vector<int> initMask = Unsafe.ReadUnaligned<Vector<int>>(
    ref Unsafe.As<int, byte>(ref MemoryMarshal.GetReference(
        (ReadOnlySpan<int>)new int[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 })));
```

I opted not to use dedicated `Vector128` and `Vector256` versions (for `.Create`) because it did not seem worth it given niche-ness of this use-case and just `Vector<T>` has good enough codegen.

Reply:
It should not really matter, but the `static readonly` field potentially introduces a static initialization (e.g. for R2R, Mono, etc) + it's more code then just inlined at a single use so I'd not use it